### PR TITLE
Verify email

### PR DESCRIPTION
**Pull request by Freshrojek**
Think this branch is ready enough.
I have updated the email sending view to now send through celery using the `tasks.py` file.
To send a verification email you need to set the appropriate settings in your environment for email, then sign up and on the `member_details` page there will be a button to send a verification email.
 If you are already verified, the button will no longer be there and you can see the `email verified` field to the member model is updated.

### CODE REVIEW

**Comment by Freshrojek** [comment]
I have a test email if anyone wants the information

File: memberships/models.py
Comment (SamWinterhalder) [inline memberships/models.py:132]: Why is this repeated?

Comment (SamWinterhalder) [inline memberships/models.py:132]: Why is this repeated?

File: memberships/views.py
Comment (SamWinterhalder) [inline memberships/views.py:202]: Commenting out `@login_required()` means that when a user who is not logged in, `AnonymousUser`, tries to go to `/memberships/details/`, an `AttributeError` is given as there is no attribute `member` for an `AnonymousUser`.

**Review by SamWinterhalder — COMMENTED** [review]
Looks great so far!
A few small things though which I think need a little bit of tweaking.

**Comment by SamWinterhalder** [comment]
One other thing - I think `.idea` should be added to `.gitignore` as from what I understand it is for your local project settings.

**Comment by SamWinterhalder** [comment]
Sorry to keep mentioning stuff but I also think that all the functions made for the email verification should be moved to its own file. `views.py` is inteded to only contain a function for each webpage.

File: memberships/models.py
Comment (Freshrojek) [reply memberships/models.py:132]: Must've been a mistake when rebasing - thanks!

**Comment by Freshrojek** [comment]
> One other thing - I think `.idea` should be added to `.gitignore` as from what I understand it is for your local project settings.

Good idea

**Comment by Freshrojek** [comment]
> Sorry to keep mentioning stuff but I also think that all the functions made for the email verification should be moved to its own file. `views.py` is inteded to only contain a function for each webpage. Or are these the views?

The more suggestions the better :) yeah it's only used once but after you mentioned it it probably would be best to have it in a file separately.

**Comment by Freshrojek** [comment]
@SamWinterhalder Actually, I'm not too sure now as they are both the views for `verify_sent.html` and `verify_confirmation.html`

File: memberships/views.py
Comment (SamWinterhalder) [inline memberships/views.py:45]: Last thing I can see is just all this commented code. It would be much clearer if this was not here.

**Review by SamWinterhalder — DISMISSED** [review]
LGTM - I think that post MVP we should look at limiting the rate/amount of time the user can request the verficiation email but that is not for now.

**Comment by Freshrojek** [comment]
Good idea, should I log a ticket for it?

**Comment by Freshrojek** [comment]
gonna have to rebase before merging because all commits need to be signed, so I might need you to look again after rebase lol

**Comment by SamWinterhalder** [comment]
Erm, there are a lot of duplicate commits 😅

Could ya `cherry-pick` please xD

**Comment by Freshrojek** [comment]
There we go, just got myself in and out of a fine hot git mess 😰

**Comment by Freshrojek** [comment]
Just added the welcome email quickly as it needs the celery set-up I have used for the verification email

**Comment by Freshrojek** [comment]
Since rebasing the celery worker receives the task but it stops there, `task_send_email` doesn't log or send an email anymore.

**Comment by Freshrojek** [comment]
and I always keep getting redirected to `confirm.html` whatever I do

**Comment by SamWinterhalder** [comment]
> and I always keep getting redirected to `confirm.html` whatever I do

See #282

**Comment by SamWinterhalder** [comment]
Emails seem to have been sent for me
image

**Comment by Freshrojek** [comment]
Nice, did the welcome email go through too?

**Comment by SamWinterhalder** [comment]
> 
> 
> Nice! Strange

Yes - if you could try getting it working on your system I can approve

**Comment by Freshrojek** [comment]
Yup just a simple restart of pycharm did it lol

**Review by SamWinterhalder — DISMISSED** [review]

**Comment by Freshrojek** [comment]
Don't merge this yet, it needs a few updates since I merged Master into the branch

**Comment by SamWinterhalder** [comment]
It seemed to be working fine when I approved it (and it was up to date)

**Comment by Freshrojek** [comment]
Yeah just got to remove the email verified field  from the member settings form I think

**Comment by Freshrojek** [comment]
@SamWinterhalder If ya have time could ya take a look at this please?

**Review by jamesgeddes — DISMISSED** [review]

**Comment by jamesgeddes** [comment]
@Freshrojek approved, rebase and merge :+1:

**Comment by Freshrojek** [comment]
> @SamWinterhalder If ya have time could ya take a look at this please?
> 

Yeah no worries at all I feel your pain 💀

**Comment by Freshrojek** [comment]
> Sorry mate been real busy, could @jamesgeddes review please?

Yeah no worries at all I feel your pain 💀

**Comment by Freshrojek** [comment]
all rebased @jamesgeddes :)

**Review by jamesgeddes — APPROVED** [review]

**Pull request merged**
